traverson4j-hc5/src/main/java/uk/co/autotrader/traverson/http/ApacheHttpConverters.java

```diff
@@ -62,14 +62,20 @@ public ClassicHttpRequest toRequest(Request request) {
         Response<T> response = new Response<>();
         response.setUri(uri);
         response.setStatusCode(httpResponse.getCode());
-        for (Header responseHeader : httpResponse.getHeaders()) {
-            response.addResponseHeader(responseHeader.getName(), responseHeader.getValue());
-        }
+        Arrays.asList(httpResponse.getHeaders()).forEach(header ->
+                response.addResponseHeader(header.getName(), header.getValue()));
 
         HttpEntity httpEntity = httpResponse.getEntity();
         if (httpEntity != null) {
-            InputStream content = httpEntity.getContent();
-            response.setResource(conversionService.convert(content, returnType));
+            if (httpResponse.getCode() >= 200 && httpResponse.getCode() < 300) {
+                response.setResource(conversionService.convert(httpEntity.getContent(), returnType));
+            } else {
+                try {
+                    response.setError(conversionService.convert(httpEntity.getContent(), String.class));
+                } catch (ConversionException ignore) {
```

Comment:
Not 100% sure what will happen now with incomplete Traversal cases, will this no longer throw the incomplete traversal exception? If so what do you propose for clients to do to determine this?